src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -477,13 +478,15 @@ int get_model_prefer_threads(const int num_streams,
                 model_prefer = proc_type_table[0][ALL_PROC];
             }
 #else
-            bool fp_intesive = !ov::op::util::has_op_with_type<ov::op::v0::FakeQuantize>(model);
+            bool fp_intensive = has_matmul_with_compressed_weights(model);
             const int int8_threshold = 4;  // ~relative efficiency of the VNNI-intensive code for Big vs Little cores;
             const int fp32_threshold = 2;  // ~relative efficiency of the AVX2 fp32 code for Big vs Little cores;
             // by default the latency case uses (faster) Big cores only, depending on the compute ratio
             model_prefer = proc_type_table[0][MAIN_CORE_PROC] > (proc_type_table[0][EFFICIENT_CORE_PROC] /
-                                                                 (fp_intesive ? fp32_threshold : int8_threshold))
-                               ? proc_type_table[0][MAIN_CORE_PROC]
+                                                                 (fp_intensive ? int8_threshold : fp32_threshold))
+                               ? ((!fp_intensive && ov::get_number_of_blocked_cores())
```

Comment:
Can we extract MTL case as separate one on line 491 to avoid impact on original condition.
I still think the proper way is to adjust int8/fp32 thresholds but current solition still can be accepted. 
So for MTL lets check that model `has_matmul_with_compressed_weights` and use P cores only in that case. 
Please also add comment that ov::get_number_of_blocked_cores() is actaully used to detect MTL